river/command/filter.zig

```diff
@@ -23,14 +23,25 @@ const util = @import("../util.zig");
 const Error = @import("../command.zig").Error;
 const Seat = @import("../Seat.zig");
 
-fn appendFilter(
+fn modifyFilter(
     allocator: *std.mem.Allocator,
     list: *std.ArrayList([]const u8),
     args: []const []const u8,
+    operation: enum { add, remove },
 ) Error!void {
     if (args.len < 2) return Error.NotEnoughArguments;
     if (args.len > 2) return Error.TooManyArguments;
-    try list.append(try std.mem.dupe(allocator, u8, args[1]));
+    for (list.items) |*filter, i| {
+        if (std.mem.eql(u8, filter.*, args[1])) {
+            if (operation == .remove) {
+                allocator.free(list.orderedRemove(i));
+            }
+            return;
+        }
+    }
+    if (operation == .add) {
+        try list.append(try std.mem.dupe(allocator, u8, args[1]));
```

Comment:
This will leak memory currently if the append() fails bu the dupe doesn't. To fix this we can first call `ensureUnususedCapacity(1)` and then use `appendAssumeCapacity()`.

Reply:
Actually, according to `git blame` it **is** my fault :D